lib/gem2rpm/context.rb

```diff
@@ -0,0 +1,54 @@
+require 'gem2rpm/template_helpers'
+
+module Gem2Rpm
+  class Context
+    include Gem2Rpm::TemplateHelpers
+
+    attr_reader :nongem, :local, :doc_subpackage, :package, :format, :spec,
+      :config, :runtime_dependencies, :development_dependencies, :tests,
+      :files, :download_path
+
+    def initialize(fname, nongem = true, local = false, doc_subpackage = true)
+      @nongem = nongem
+      @local = local
+      @doc_subpackage = doc_subpackage
+
+      @package = Gem2Rpm::Package.new(fname)
+      # Deprecate, kept just for backward compatibility.
+      @format = Gem2Rpm::Format.new(@package)
+      @spec = Gem2Rpm::Specification.new(@package.spec)
+
+      @config = Configuration.instance.reset
+
+      @runtime_dependencies = Gem2Rpm::RpmDependencyList.new(@spec.runtime_dependencies)
+      @development_dependencies = Gem2Rpm::RpmDependencyList.new(@spec.development_dependencies)
+
+      @tests = TestSuite.new(spec)
+
+      # Ruby 2.0 doesn't have sorted files
+      @files = RpmFileList.new(spec.files.sort)
```

Comment:
I think this is the workaround you meant

Reply:
This is one of the places where the sort could be applied. However, this might impact Ruby 2.0 users, because it will change the file order everywhere. Therefore I think that it would be less intrusive if you sort the file list just in the test template, e.g. from:

~~~
<%= main_files.to_rpm %>
~~~

to

~~~
<%= main_files.sort.to_rpm %>
~~~

IOW it would "fix" just the test case. But you can convince me that the stable output of gem2rpm between Ruby versions is more important then the stable output between versions :)

Reply:
An argument can be made for both. Personally I don't care for Ruby 2.0 and sorting only in tests gives a (trivial) performance speedup at runtime.

Reply:
Ok, it does appear that your suggestion doesn't work: `NoMethodError: undefined method `to_rpm' for [".travis.yml", "exe", "ext", "lib"]:Array`

So I'm going back to sorting `spec.files`. You could argue it makes `gem2rpm` a predictable tool, regardless of the Ruby version.

Reply:
~~~
NoMethodError: undefined method `to_rpm' for [".travis.yml", "exe", "ext", "lib"]:Array
~~~

:crying_cat_face: 

But sorting the output of `to_rpm` could also do the job, couldn't it?

Reply:
Then it gets sorted by the full line, like `%exclude`. Right now I think we're testing closer to real application behavior and make sure Ruby 2.0 behaves the same as Ruby 2.1+. I'm starting to lean to that.

Reply:
`to_rpm.sort` does not work, because `to_rpm` is just string, that would need some conversion again. So will probably stick with this. So far nobody really complained about the order, so hopefully it stays like that :) Thx for the effort